modules/mixercat/Pair.js

```diff
@@ -0,0 +1,72 @@
+'use strict';
+
+const Chooser = require('../choose/Chooser');
+const chooser = new Chooser();
+
+module.exports = class Pair {
+  async pairMembers(members, mixerCatModel) {    
+    const matches = [];
+    const userMap = {};
+    for (let i=0; i < members.length; i++) {
+      const user = members[i];    
+      if (userMap[user]) {        
+        continue;
+      }
+      userMap[user] = true;
+      const pairedUsers = await this.getPair(user, members, mixerCatModel);              
+      if (pairedUsers.length) {       
+        userMap[pairedUsers[1]] = true;        
+        matches.push(pairedUsers);                
+      } 
+    }
+
+    return matches;
+  }
+  async getPair(userId, users, pairedTable) {  
+    const potentialPairs = await this.getValidPairs(userId, users, pairedTable);
+    if (!potentialPairs.length) {
+      // User is out of matches.
+      return [];
+    }
+
+    const pairedMember = chooser.getRandomUser(potentialPairs);
+    const result = await this.savePair(userId, pairedMember, pairedTable);
+    
+    return [userId, pairedMember];
+  }
+
+  async getValidPairs(userId, users, pairedTable) {
+    const previousPairs = await pairedTable
+      .findAll({
+        where: {
+          user_id: userId,          
+        },
+      })
+      .map(it => {
+        return it.get('paired_with');
+      });
+   
+    return users.filter(it => {
+      return !previousPairs.includes(it) && it !== userId;
+    });
```

Comment:
Not sure if this is your intent but as written, this will eventually stop pairing after having paired everyone at least once. You may want to include some kind of date column to check as well to expire old pairings.